pkg/api/kptfile/v1/types.go

```diff
@@ -0,0 +1,316 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package defines the schema for Kptfile version v1.
+package v1
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	KptFileName       = "Kptfile"
+	KptFileKind       = "Kptfile"
+	KptFileGroup      = "kpt.dev"
+	KptFileVersion    = "v1"
+	KptFileAPIVersion = KptFileGroup + "/" + KptFileVersion
+)
+
+// TypeMeta is the TypeMeta for KptFile instances.
+var TypeMeta = yaml.ResourceMeta{
+	TypeMeta: yaml.TypeMeta{
+		APIVersion: KptFileAPIVersion,
+		Kind:       KptFileName,
+	},
+}
+
+// KptFile contains information about a package managed with kpt.
+type KptFile struct {
+	yaml.ResourceMeta `yaml:",inline"`
+
+	Upstream *Upstream `yaml:"upstream,omitempty"`
+
+	// UpstreamLock is a resolved locator for the last fetch of the package.
+	UpstreamLock *UpstreamLock `yaml:"upstreamLock,omitempty"`
+
+	// Info contains metadata such as license, documentation, etc.
+	Info *PackageInfo `yaml:"info,omitempty"`
+
+	// Pipeline declares the pipeline of functions.
+	Pipeline *Pipeline `yaml:"pipeline,omitempty"`
+
+	// Inventory contains parameters for the inventory object used in apply.
+	Inventory *Inventory `yaml:"inventory,omitempty"`
+
+	// Ignore contains a list of .gitignore patterns for files and directories
+	// that should be ignored by kpt. Files/directories that match will be
+	// fetched and updated as part of the package, but will be ignored by the other
+	// kpt functionality. The patterns will only impact resources within the
+	// same package, and it is not possible to ignore subpackages.
+	//
+	// The following patterns are supported:
+	// * Absolute path (/path/to/ignore): If the pattern has a leading slash, it will
+	//   only match the specific path underneath the package root.
+	//   Example: "/ignore.me" will only match the file if it is in the root of
+	//   the package.
+	// * Relative path (path/to/ignore): If the pattern does not have a leading slash,
+	//   it will match at any depth in the package.
+	//   Example: "ignore.me" will match all files called "ignore.me" within the
+	//   package.
+	// * Accept pattern (!path/to/accept): Negate a previous pattern so the path
+	//   will be included.
+	// * Directory pattern (path/to/directory/): If the pattern has a slash at the
+	//   end, it will only match directories.
+	// * Glob pattern (path/to/*.txt): * matches all characters except slash.
+	//
+	// Recursive patterns (path/**/ignore) are not supported.
+	Ignore []string `yaml:"ignore,omitempty"`
```

Comment:
@mortent you're ok with this being in v1?

Reply:
I think we should just back it out. It would be an unused field, so we should add it back in when we are ready with the full implementation.

Reply:
Yeah, I would also suggest removing it.

Reply:
done